rfd/00xx-snowflake-support.md

```diff
@@ -0,0 +1,170 @@
+---
+authors: Jakub Nyckowski (jakub.nyckowski@goteleport.com)
+state: draft
+---
+
+# RFD XX - Snowflake Database support
+
+## What
+
+[Snowflake](https://www.snowflake.com/) integration in Teleport Database Access. 
+
+## Why
+
+We want to increase the number of databases supported by Teleport. 
+
+## Details
+
+The Snowflake support can be split into 4 parts:
+
+1. Database access support - changes inside Teleport Data Access.
+2. CLI support - `snowsql` the official Snowflake client.
+3. GUI clients support - DataGrip by JetBrains, DBeaver and similar.
+4. WebUI integration https://app.snowflake.com/.
+
+### Snowflake architecture
+
+* Snowflake uses REST API for communication. 
+* Payload is a regular JSON. Some queries can also carry payload in [Arrow format](https://arrow.apache.org/). 
+* Snowflake uses a slightly modified JWT token flow for authentication (details below).
+
+### Teleport configuration
+
+Like in case of other databases, a user will have to add an entry to the teleport configuration file. Example:
+
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+      - name: "snowflake"
+      description: "Snowflake database"
+      protocol: "snowflake"
+      uri: "im12345.us-east-2.aws.snowflakecomputing.com"
+```
+
+The next step would be to add teleport CA to the Snowflake user account. `tctl` could learn to export the required keys 
+in the same way as it does today for different databases (`tctl auth sign` command). The exported key must then be added 
+to Snowflake by an administrator. The procedure of adding the key to Snowflake is described [here](https://docs.snowflake.com/en/user-guide/key-pair-auth.html#step-4-assign-the-public-key-to-a-snowflake-user).
+
+### Database access support
+
+`tsh db login` should export a user private key in PCKS8 format (that's the only format supported by `snowsql` and some SDKs).
+That key should be used to sign a "Teleport JWT". Client then should send the request with generated JWT to authenticate 
+with Snowflake. Teleport can intercept HTTP communication between a client and the Snowflake. The first request send by 
+the client should be to the `/session/v1/login-request` endpoint which contains all login information as username, JWT. 
+This data can be used to authenticate user in Teleport. JWT token prepared by the client can be then read and replaces by Snowflake JWT
```

Comment:
This is not super clear. Since this is HTTP, does the client send authentication info (JWT) with every request? What does "authenticate user in Teleport" mean?